### DISCUSSION TITLE
امن ترین پورت

### DISCUSSION THREAD

**Discussion by arman2012** [hiddify/Hiddify-Manager#179] [Q&A سوالات رایج]
سلام میخواستم بدونم vless امن تره یا trojan و اینکه دامین فرانتینگ امن تره یا مستقیم با cdn?

**Comment by hiddify-com** [comment]
امنیت فرقی نداره همه امن هستند وقتی روی tls هستند
وقتی روی tls نیستند اگر encryption داشته باشند امن هستند
حتی اگر encryption هم نداشته باشند باز هم امن هست چون سایت ها خودشون از https استفاده میکنند

اگر موضوعتون کشف نشدن توسط فیلترچی هست 
دامین فرانتینگ برای اینه که هیچ اثری به جا نذاره 
تو cdn دامنه مشخص هست
و توی حالت دایرکت هم دامنه هم cdn مشخصه

**Reply by robin98** [reply]
@hiddify 

> دامین فرانتینگ برای اینه که هیچ اثری به جا نذاره

با این توصیف فکر میکنم یا من اشتباه می کنم، یا معماری دامین فرانتینگ مشکلی داره.

ببینید الان من یک دامنه رو از `تنظیمات > دامین فرانتینگ` اد کردم که ساب دامینی از دامنه خودم هست، که داخل A رکورد اون یک IP سالم از CF وجود داره.
حالا وقتی از داخل پنل کاربری میام و اون لینکی که توی اسمش Fake داره رو کپی و ایمپورت می کنم، در قسمت Camouflage Domain اون، url دایرکتِ سرورم پیست شده و این یعنی **عملا آی پی اصلی سرور رو داره اعلام میکنه**؛ و طبق پیشفرض ذهنی من، مقدار Camouflage به شکل clear text هست.
اما [اگر اطلاعاتم صحیح باشه] این مقدارِ SNI هست که در شرایط خاصی (فکر کنم فعال بودن uTLS) انکریپت میشه.



پس با این تفاسیر، استفاده از مود دامین فرانتینگ یا همون کانکشن های Fake اتفاقا نه تنها دامین رو لو میده بلکه عملا داره IP اصلی سرور رو هم لو میده.
و این دقیقا متضاد اون چیزی هست که از دامین فرانتینگ یا همین توضیحاتی که نوشتید انتظار میره.

به هر حال چیزی که تا این لحظه قطعیه، تا الان و روی ورژن `0.9.57`، هر لینک `tls WS Fake x` که از پنل کپی میشه حاوی یک نشتی اطلاعات هست، در صورتی که حداقل می شد از یکی از اون لینک های CDN که اضافه کردیم استفاده می شد. (خودم دستی توی کلاینتم عوضش کردم و الان هم داره کار می کنه)

**Comment by arman2012** [comment]
https://github.com/hiddify/hiddify-config/discussions/118   در این دیسکاشن شما گفتید که دامین فرانتینگ یا همون فیک باعث تشخیص سریع تر فیلتر چی میشه
